File: cpp/28a_1560A_DislikeOfThrees.cpp
#include<bits/stdc++.h>

int main(){
    int t, k, cnt, jump;
    std::cin>>t;
    std::vector<int> ans;
    for(int i = 0; i < t; i++){
        std::cin>>k;
        cnt = 0, jump = 1;
        while(jump <= k){
            cnt++;
            while((cnt % 3) == 0 || (cnt % 10) == 3){
                cnt++;
            }
            jump++;
        }
        ans.push_back(cnt);
    }
    for(auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}

File: python/28a_1560A_DislikeOfThrees.py
def main():
    t = int(input())
    ans = []
    for _ in range(t):
        k = int(input())
        cnt, jump = 0, 1
        while jump <= k:
            cnt += 1
            while cnt % 3 == 0 or (cnt % 10) == 3:
                cnt += 1
            jump += 1
        ans.append(cnt)
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()